Sources/PostgresNIO/New/Data/Range+PostgresCodable.swift

```diff
@@ -0,0 +1,104 @@
+import NIOCore
+
+extension RangeExpression where Bound: PostgresRangeBound {
+    public static var psqlFormat: PostgresFormat { return .binary }
+}
+
+// MARK: Range
+
+extension Range: PostgresEncodable where Bound: PostgresRangeBound {
+    public static var psqlType: PostgresDataType { return Bound.rangeType }
+
+    @inlinable
+    public func encode<JSONEncoder: PostgresJSONEncoder>(
+        into byteBuffer: inout ByteBuffer,
+        context: PostgresEncodingContext<JSONEncoder>
+    ) {
+        let postgresRange = PostgresRange<Bound>(range: self)
+        postgresRange.encode(into: &byteBuffer)
+    }
+}
+
+extension Range: PostgresNonThrowingEncodable where Bound: PostgresRangeBound {}
+
+extension Range: PostgresDecodable where Bound: PostgresRangeBound {
+    @inlinable
+    public init<JSONDecoder: PostgresJSONDecoder>(
+        from buffer: inout ByteBuffer,
+        type: PostgresDataType,
+        format: PostgresFormat,
+        context: PostgresDecodingContext<JSONDecoder>
+    ) throws {
+        guard type == Bound.rangeType else {
+            throw PostgresDecodingError.Code.typeMismatch
+        }
+
+        guard let postgresRange = PostgresRange<Bound>(
+            from: &buffer,
+            format: format
+        ) else {
+            throw PostgresDecodingError.Code.failure
+        }
+
+        guard let lowerBound: Bound = postgresRange.lowerBound,
```

Comment:
Do we have to check if the range is inclusive here?

Reply:
See new commit https://github.com/vapor/postgres-nio/pull/330/commits/5aa2b1924a7af050efba4d7e05a5936cd26ecee7